### PR TITLE
Cleanup to increase readability

### PR DESCRIPTION
**Pull request by ShFil119**

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
`== false` is the preferred way to do a comparison in this codebase.

**Comment by AaronVanGeffen** [comment]
> `== false` is the preferred way to do a comparison in this codebase.

Not in general, surely. Any particular lines you would like to see reverted?

**Comment by Gymnasiast** [comment]
@ShFil119 Don't yet follow Duncan's advice, this is currently being discussed.

**Comment by ShFil119** [comment]
On gitter?

**Comment by Gymnasiast** [comment]
@ShFil119 Yes, but in the private team chat.

File: src/openrct2-ui/windows/EditorObjectSelection.cpp [hunk 1]
```diff
@@ -1463,10 +1463,7 @@ static bool filter_chunks(const ObjectRepositoryItem* item)
                     break;
                 }
             }
-            if (_filter_flags & (1 << (gRideCategories[rideType] + _numSourceGameItems)))
-                return true;
-
-            return false;
+            return (_filter_flags & (1 << (gRideCategories[rideType] + _numSourceGameItems))) != 0;
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/EditorObjectSelection.cpp]: I prefer the old version, I think that is more readable than the new one.

File: src/openrct2-ui/windows/RideConstruction.cpp [hunk 2]
```diff
@@ -2441,14 +2441,14 @@ static void sub_6CBCE2(
         // Set the temporary track element
         _tempTrackTileElement.SetType(TILE_ELEMENT_TYPE_TRACK);
         _tempTrackTileElement.SetDirection(trackDirection);
-        _tempTrackTileElement.AsTrack()->SetHasChain((edx & 0x10000) ? true : false);
+        _tempTrackTileElement.AsTrack()->SetHasChain((edx & 0x10000) != 0);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/RideConstruction.cpp]: I find the old notation easier to read.

File: src/openrct2/ReplayManager.cpp [hunk 3]
```diff
@@ -610,7 +610,7 @@ namespace OpenRCT2
                 return false;
 
             char buffer[128];
-            while (feof(fp) == false)
+            while (!static_cast<bool>(feof(fp)))
```
Comment (Gymnasiast) [inline src/openrct2/ReplayManager.cpp]: Not sure about this one.

File: src/openrct2/ReplayManager.cpp [hunk 4]
```diff
@@ -774,10 +774,7 @@ namespace OpenRCT2
 
         bool Compatible(ReplayRecordData& data)
         {
-            if (data.version == 1 && ReplayVersion == 2)
-                return true;
-
-            return false;
+            return data.version == 1 && ReplayVersion == 2;
```
Comment (Gymnasiast) [inline src/openrct2/ReplayManager.cpp]: I have a slight preference towards the old notation, but at least this is as easy to read.

File: src/openrct2/interface/InteractiveConsole.cpp [hunk 5]
```diff
@@ -1379,9 +1379,7 @@ static int32_t cc_replay_stoprecord(InteractiveConsole& console, const arguments
     }
 
     auto* replayManager = OpenRCT2::GetContext()->GetReplayManager();
-    if (replayManager->IsRecording() == false && replayManager->IsNormalising() == false)
-    {
-        console.WriteFormatLine("Replay currently not recording");
+    if (replayMan!replayManager->IsRecording()yMan!replayManager->IsNormalising()       console.WriteFormatLine("Replay currently not recording");
```
Comment (Gymnasiast) [inline src/openrct2/interface/InteractiveConsole.cpp]: This looks like an unintentional change.

File: src/openrct2/management/Finance.cpp [hunk 6]
```diff
@@ -91,11 +91,9 @@ bool finance_check_affordability(money32 cost, uint32_t flags)
 {
     if (cost <= 0)
         return true;
-    if (finance_check_money_required(flags) == false)
+    if (!finance_check_money_required(flags))
         return true;
-    if (cost > gCash)
-        return false;
-    return true;
+    return cost <= gCash;
```
Comment (Gymnasiast) [inline src/openrct2/management/Finance.cpp]: I prefer the old notation.

File: src/openrct2/network/Network.cpp [hunk 7]
```diff
@@ -2087,8 +2087,7 @@ NetworkPlayer* Network::AddPlayer(const std::string& name, const std::string& ke
             if (networkUser == nullptr)
             {
                 player->Group = GetDefaultGroup();
-                if (name.empty() == false)
-                {
+                if (name.e!name.empty()            {
```
Comment (Gymnasiast) [inline src/openrct2/network/Network.cpp]: Should be reverted.

File: src/openrct2-ui/windows/RideConstruction.cpp (hunk 2 above)
Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/RideConstruction.cpp]: I'm inclined to disagree. This sort of boolean expression does not warrant a ternary operator, imo.

File: src/openrct2/paint/Paint.cpp [hunk 8]
```diff
@@ -306,42 +306,26 @@ static bool check_bounding_box(const paint_struct_bound_box& initialBBox, const
...
 template<> bool check_bounding_box<0>(const paint_struct_bound_box& initialBBox, const paint_struct_bound_box& currentBBox)
 {
-    if (initialBBox.z_end >= currentBBox.z && initialBBox.y_end >= currentBBox.y && initialBBox.x_end >= currentBBox.x
-        && !(initialBBox.z < currentBBox.z_end && initialBBox.y < currentBBox.y_end && initialBBox.x < currentBBox.x_end))
-    {
-        return true;
-    }
-    return false;
+    return initialBBox.z_end >= currentBBox.z && initialBBox.y_end >= currentBBox.y && initialBBox.x_end >= currentBBox.x
+        && !(initialBBox.z < currentBBox.z_end && initialBBox.y < currentBBox.y_end && initialBBox.x < currentBBox.x_end);
```
Comment (Gymnasiast) [inline src/openrct2/paint/Paint.cpp]: All of these are slightly easier to read in their old notation (though still awful).

File: src/openrct2/peep/Guest.cpp [hunk 9]
```diff
@@ -5901,12 +5901,7 @@ bool Guest::ShouldFindBench()
         return false;
     }
 
-    if (!GetNextIsSurface() && !GetNextIsSloped())
-    {
-        return true;
-    }
-
-    return false;
+    return !GetNextIsSurface() && !GetNextIsSloped();
```
Comment (Gymnasiast) [inline src/openrct2/peep/Guest.cpp]: Old notation is easier to read.

File: src/openrct2/platform/Posix.cpp [hunk 10]
```diff
@@ -130,11 +130,7 @@ bool platform_directory_exists(const utf8* path)
     struct stat dirinfo;
     int32_t result = stat(buffer, &dirinfo);
     log_verbose("checking dir %s, result = %d, is_dir = %d", buffer, result, S_ISDIR(dirinfo.st_mode));
-    if ((result != 0) || !S_ISDIR(dirinfo.st_mode))
-    {
-        return false;
-    }
-    return true;
+    return !((result != 0) || !S_ISDIR(dirinfo.st_mode));
```
Comment (Gymnasiast) [inline src/openrct2/platform/Posix.cpp]: Old notation is easier to read.

File: src/openrct2/ReplayManager.cpp (hunk 3 above)
Comment (AaronVanGeffen) [reply src/openrct2/ReplayManager.cpp]: I, too, prefer the old style, but for the proposed solution is more in line with the C++ way. (Of course, better solution would be to not use a libc function.)

File: src/openrct2/ride/Ride.cpp [hunk 11]
```diff
@@ -2573,12 +2573,7 @@ bool Ride::CanBreakDown() const
     }
 
     rct_ride_entry* entry = get_ride_entry_by_ride(this);
-    if (entry == nullptr || entry->flags & RIDE_ENTRY_FLAG_CANNOT_BREAK_DOWN)
-    {
-        return false;
-    }
-
-    return true;
+    return !(entry == nullptr || entry->flags & RIDE_ENTRY_FLAG_CANNOT_BREAK_DOWN);
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp]: Ditto.

File: src/openrct2/ride/Ride.cpp [hunk 12]
```diff
@@ -7773,13 +7768,9 @@ uint8_t ride_entry_get_first_non_null_ride_type(const rct_ride_entry* rideEntry)
...
 {
-    if (rideType == RIDE_TYPE_LOOPING_ROLLER_COASTER || rideType == RIDE_TYPE_CORKSCREW_ROLLER_COASTER
+    return rideType == RIDE_TYPE_LOOPING_ROLLER_COASTER || rideType == RIDE_TYPE_CORKSCREW_ROLLER_COASTER
         || rideType == RIDE_TYPE_TWISTER_ROLLER_COASTER || rideType == RIDE_TYPE_VERTICAL_DROP_ROLLER_COASTER
-        || rideType == RIDE_TYPE_GIGA_COASTER || rideType == RIDE_TYPE_JUNIOR_ROLLER_COASTER)
-    {
-        return true;
-    }
-    return false;
+        || rideType == RIDE_TYPE_GIGA_COASTER || rideType == RIDE_TYPE_JUNIOR_ROLLER_COASTER;
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp]: Ditto.

File: src/openrct2/ReplayManager.cpp (hunk 4 above)
Comment (AaronVanGeffen) [reply src/openrct2/ReplayManager.cpp]: I prefer to return single boolean expressions as oneliners, as proposed.

File: src/openrct2/world/Map.cpp [hunk 13]
```diff
@@ -706,12 +706,8 @@ int32_t map_height_from_slope(const CoordsXY coords, int32_t slope, bool isSlope
...
 bool map_is_location_valid(const CoordsXY coords)
 {
-    if (coords.x < (MAXIMUM_MAP_SIZE_TECHNICAL * 32) && coords.x >= 0 && coords.y < (MAXIMUM_MAP_SIZE_TECHNICAL * 32)
-        && coords.y >= 0)
-    {
-        return true;
-    }
-    return false;
+    return coords.x < (MAXIMUM_MAP_SIZE_TECHNICAL * 32) && coords.x >= 0 && coords.y < (MAXIMUM_MAP_SIZE_TECHNICAL * 32)
+        && coords.y >= 0;
```
Comment (Gymnasiast) [inline src/openrct2/world/Map.cpp]: Old notation is easier to read.

File: src/openrct2/world/Sprite.cpp [hunk 14]
```diff
@@ -712,10 +712,7 @@ static bool litter_can_be_at(int32_t x, int32_t y, int32_t z)
         if (pathZ < z || pathZ >= z + 32)
             continue;
 
-        if (tile_element_is_underground(tileElement))
-            return false;
-
-        return true;
+        return !tile_element_is_underground(tileElement);
```
Comment (Gymnasiast) [inline src/openrct2/world/Sprite.cpp]: Ditto.

**Comment by Gymnasiast** [comment]
Okay, this `bool == false` vs. `!bool` has been discussed. We have opted for the latter, so you keep those changes.

File: src/openrct2-ui/windows/RideConstruction.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2-ui/windows/RideConstruction.cpp]: I'm not very fussed, really. Let's keep @ShFil119 's way, then.

**Comment by ShFil119** [comment]
Fixed code breaking builds.

File: src/openrct2/platform/Posix.cpp (hunk 10 above)
Comment (ShFil119) [reply src/openrct2/platform/Posix.cpp]: `return result == 0) && S_ISDIR(dirinfo.st_mode);`?

File: src/openrct2/management/Finance.cpp (hunk 6 above)
Comment (ShFil119) [reply src/openrct2/management/Finance.cpp]: tweaked a bit in another way

File: src/openrct2/platform/Posix.cpp (hunk 10 above)
Comment (AaronVanGeffen) [reply src/openrct2/platform/Posix.cpp]:
```suggestion
    return result == 0 && S_ISDIR(dirinfo.st_mode);
```

https://en.wikipedia.org/wiki/De_Morgan%27s_laws

File: src/openrct2/ride/Ride.cpp (hunk 11 above)
Comment (AaronVanGeffen) [reply src/openrct2/ride/Ride.cpp]:
```suggestion
    return entry != nullptr && entry->flags & RIDE_ENTRY_FLAG_CANNOT_BREAK_DOWN;
```

https://en.wikipedia.org/wiki/De_Morgan%27s_laws

**Comment by AaronVanGeffen** [comment]
I prefer the proposed changes to return expressions directly rather than if/else-ing them into trivial boolean returns. However, I do think some of the expression can be improved for readability. I've left some comments to that end.

File: src/openrct2/ride/Ride.cpp (hunk 11 above)
Comment (ShFil119) [reply src/openrct2/ride/Ride.cpp]: `return entry != nullptr && (entry->flags & RIDE_ENTRY_FLAG_CANNOT_BREAK_DOWN) == 0;`?

File: src/openrct2/world/Map.cpp (hunk 13 above)
Comment (ZehMatt) [reply src/openrct2/world/Map.cpp]: I would even suggest to split this into multiple tests.

**Comment by Gymnasiast** [comment]
I took the changes we agreed with and opened #8899 (putting you as author).

**Comment by AaronVanGeffen** [comment]
When amended, I believe 9d56a9245383b1b66daccb4c128462c1909d7085 (currently the second commit in this PR) would make a nice improvement as well.

@ShFil119, would you like to finish this? Otherwise, I'll happily pick it up.

**Comment by Gymnasiast** [comment]
This PR has not been updated in more than a month, with no replies to our comments. I doubt reopening it will achieve anything. But I'd love to be proven wrong.

File: src/openrct2/ReplayManager.cpp (hunk 3 above)
Comment (ShFil119) [reply src/openrct2/ReplayManager.cpp]: `feof(fp) == 0`?

File: src/openrct2/peep/Guest.cpp (hunk 9 above)
Comment (ShFil119) [reply src/openrct2/peep/Guest.cpp]: `!(GetNextIsSurface() || GetNextIsSloped())`?

Comment (AaronVanGeffen) [reply src/openrct2/peep/Guest.cpp]: I'm fairly sure @Gymnasiast referred to the if-block. Personally, I disagree; `!GetNextIsSurface() && !GetNextIsSloped()` is a succinct and equivalent expression.

File: src/openrct2/world/Map.cpp (hunk 13 above)
Comment (ShFil119) [reply src/openrct2/world/Map.cpp]: done

File: src/openrct2/paint/Paint.cpp (hunk 8 above)
Comment (AaronVanGeffen) [reply src/openrct2/paint/Paint.cpp]: Perhaps split the bounding box expressions into three: a separate check for X, Y, and Z. Then, in the end, return the conjunction of the three.

File: src/openrct2/world/Map.cpp [hunk 15]
```diff
@@ -732,12 +732,9 @@ int32_t map_height_from_slope(const CoordsXY coords, int32_t slope, bool isSlope
...
 {
-    if (coords.x < (MAXIMUM_MAP_SIZE_TECHNICAL * 32) && coords.x >= 0 && coords.y < (MAXIMUM_MAP_SIZE_TECHNICAL * 32)
-        && coords.y >= 0)
-    {
-        return true;
-    }
-    return false;
+    const bool is_x_valid = coords.x < (MAXIMUM_MAP_SIZE_TECHNICAL * 32) && coords.x >= 0;
+    const bool is_y_valid = coords.y < (MAXIMUM_MAP_SIZE_TECHNICAL * 32) && coords.y >= 0;
+    return is_x_valid && is_y_valid;
```
Comment (AaronVanGeffen) [inline src/openrct2/world/Map.cpp]: This turned out quite nice! :+1:

File: src/openrct2/interface/InteractiveConsole.cpp [hunk 16]
```diff
@@ -1445,7 +1445,7 @@ static int32_t cc_replay_stoprecord(InteractiveConsole& console, const arguments
     }
 
     auto* replayManager = OpenRCT2::GetContext()->GetReplayManager();
-    if (replayManager->IsRecording() == false && replayManager->IsNormalising() == false)
+    if (replayManager->IsRecording() && replayManager->IsNormalising())
```
Comment (Gymnasiast) [inline src/openrct2/interface/InteractiveConsole.cpp]: This reverses the old logic.

File: src/openrct2-ui/windows/EditorObjectSelection.cpp (hunk 1 above)
Comment (ShFil119) [reply src/openrct2-ui/windows/EditorObjectSelection.cpp]: Revert to original or use `return (_filter_flags & (1 << (gRideCategories[rideType] + _numSourceGameItems)));`?

File: src/openrct2/interface/InteractiveConsole.cpp (hunk 16 above)
Comment (ShFil119) [reply src/openrct2/interface/InteractiveConsole.cpp]: should be fixed now

File: src/openrct2-ui/windows/EditorObjectSelection.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2-ui/windows/EditorObjectSelection.cpp]: To the original.

File: src/openrct2/ReplayManager.cpp (hunk 4 above)
Comment (Gymnasiast) [reply src/openrct2/ReplayManager.cpp]: Right. This case seems readable to me at least.

File: src/openrct2/paint/Paint.cpp (hunk 8 above)
Comment (Gymnasiast) [reply src/openrct2/paint/Paint.cpp]: Yeah, let's go with that.

File: src/openrct2/ride/Ride.cpp (hunk 12 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Ride.cpp]: I can accept this.

File: src/openrct2/management/Finance.cpp [hunk 17]
```diff
@@ -89,13 +89,8 @@ bool finance_check_money_required(uint32_t flags)
...
 {
-    if (cost <= 0)
-        return true;
-    if (finance_check_money_required(flags) == false)
-        return true;
-    if (cost > gCash)
-        return false;
-    return true;
+    return cost <= 0 || !finance_check_money_required(flags) || cost <= gCash;
+
```
Comment (Gymnasiast) [inline src/openrct2/management/Finance.cpp]: Clang-format wants you to drop this blank line ;-)

**Comment by Gymnasiast** [comment]
There are now two points left to address. @ShFil119 , could you address these points, so that we can merge this PR?

**Comment by Gymnasiast** [comment]
You left one point unaddressed. Please address it, so we can get this merged.

File: src/openrct2/paint/Paint.cpp (hunk 8 above)
Comment (ShFil119) [reply src/openrct2/paint/Paint.cpp]: As discussed on gitter, I've reverted it for now.

**Pull request merged**